api/v1beta1/azuremachine_types.go

```diff
@@ -132,6 +132,12 @@ type AzureMachineSpec struct {
 	// +optional
 	DNSServers []string `json:"dnsServers,omitempty"`
 
+	// DisableBootstrappingVMExtension disables CAPZ Bootstrapping VM extension.
+	// Its role is to detect and report Kubernetes bootstrap failure or success.
+	// Use this setting if VMExtensions are not supported by your image.
```

Comment:
```suggestion
	// Use this setting only if VMExtensions are not supported by your image.
```
I think we want to change the wording here since it's not recommended to disable the extension. If there is another reason why someone would want to disable it then it's okay to leave it as is!